Replace react-scroll with Bootstrap 4 scrollspy

thinkh opened the pull request:
Fixes https://github.com/Caleydo/tdp_bi_bioinfodb/issues/1188

### Summary

* Remove navbar fixed and change overflow behavior in a way that the start menu itself is scrollable and not the whole page
* Replace react-scroll with [Bootstrap 4 scrollspy](https://getbootstrap.com/docs/4.6/components/scrollspy/) (which comes already with phovea_ui)
   * The BS4 scrollspy requires some `id` attributes which are kinda hard-coded following the pattern `${tabId}-tab-scrollspy-nav`
   * The [BS4 scrollspy must be refreshed](https://getbootstrap.com/docs/4.6/components/scrollspy/#scrollspyrefresh) when switching the tab
   * Jumping to the selected section/card is handled by a custom `onClick` function which scrolls the element into the view

### Screenshots

![ordino-bs4-scrollspy](https://user-images.githubusercontent.com/5851088/113211213-80859880-9275-11eb-8d4c-7a4a9e30e208.gif)

oltionchampari reviewed (changes requested):
1. So the one issue that remains is when opening the tab for the first time the current button is not active 

![grafik_2](https://user-images.githubusercontent.com/51322092/113297666-2e875600-92fb-11eb-9c65-1e28fd49e275.png)

2. There is this weird jump behavior, the active state is triggered by the button and also the scrolling.

![grafik](https://user-images.githubusercontent.com/51322092/113297813-62627b80-92fb-11eb-8c43-94e8a20367f2.gif)

thinkh commented:
Thanks for your feedback and I agree with both.

> 1. The one issue that remains is when opening the tab for the first time, the current button is not active.

This is difficult to fix, since the first element does not start at the top of the container. Hence, I would need to create a workaround, which I could not find in a reasonable amount of time. I suggest to come back to this on a later time point.

> 2. There is this weird jump behavior, the active state is triggered by the button and also the scrolling itself

This was also a tricky one, because the problem was not the Bootstrap Scrollspy, but the `ListGroup.Item` of react-bootstrap. The react element comes with some "custom" logic that it highlights the link already on click. The flickering is then caused with the `scrollIntoView` and the scrollspy that highlights each navigation links while scrolling.

As a workaround, I used plain HTML links and added the respective Bootstrap CSS classes. I also added some source code comments to avoid later confusion.
